app/scripts/modules/core/src/pipeline/config/triggers/Trigger.tsx

```diff
@@ -34,20 +40,32 @@ export class Trigger extends React.Component<ITriggerProps> {
   };
 
   private handleTriggerEnabled = () => {
-    const enabled = !this.props.trigger.enabled;
-    this.triggerUpdated({ enabled });
+    const enabled = !this.state.trigger.enabled;
+    this.updateTriggerFields({ enabled });
   };
 
   private handleTypeChange = (option: Option<string>) => {
-    const type = option.value;
-    this.triggerUpdated({ type });
-    if (this.disableAutoTriggering.includes(type)) {
-      this.triggerUpdated({ enabled: false });
-    }
+    const commonFields: Array<keyof ITrigger> = [
+      'enabled',
+      'rebake',
+      'user',
+      'type',
+      'expectedArtifactIds',
+      'runAsUser',
+      'excludedArtifactTypePatterns',
+    ];
+
+    // Clear out all non-common fields when the type is changed
+    const trigger = pick(this.state.trigger, commonFields) as ITrigger;
+    trigger.enabled = this.disableAutoTriggering.includes(option.valuetype) ? false : trigger.enabled;
+    trigger.type = option.value;
+    this.setState({ trigger }, () => this.updateTriggerFields({}));
```

Comment:
It seems like we are creating two sources of truth for the trigger, one on this component's state and one in the angular model that is updated when we call `updateTriggerFields`. I wonder if we could eliminate this component's local state and just call `updateTriggerFields` here? Otherwise, I think it's a bit confusing to have an `initialTrigger` prop that is not actually guaranteed to always be the initial data, since it will presumably stay in sync with any changes we make to the angular model? Let me know if that makes sense, I may be totally off base here.

Reply:
you're absolutely right.  I'll update this